src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -300,11 +300,11 @@ class ObjectMonitor : public CHeapObj<mtInternal> {
 
   // Returns true if the specified thread owns the ObjectMonitor. Otherwise
   // returns false and throws IllegalMonitorStateException (IMSE).
-  bool      check_owner(Thread* THREAD);
+  bool      check_owner(TRAPS);
 
-  bool      enter(TRAPS);
+  bool      enter(JavaThread* current);
   void      exit(bool not_suspended, TRAPS);
```

Comment:
Shouldn't we also remove use of TRAPS for exit()? In case of unbalanced locking we don't throw IMSE, only just assert.